test/jdk/java/awt/Frame/MegaIconTest/MegaIconTest.java

```diff
@@ -0,0 +1,271 @@
+    public IconTestButtons(IconTestFrame[] iconTests) {
+        IconTestFrame tempTest;
+        Button newBtn;
+        Panel newPnl;
+        DoneLabel newLbl;
+
+        setTitle("MegaIconTest");
+
+        setLayout(new GridLayout(iconTests.length, 1));
+
+        //For each icon test frame
+        //Get name, add button with name and action to
+        //display the window, and add label "done" after
+
+        for (int i = 0; i < iconTests.length; i++) {
+            tempTest = iconTests[i];
+            newBtn = new Button(tempTest.getTestName());
+            newLbl = new DoneLabel();
+            newBtn.addActionListener(new IconTestActionListener(tempTest,
+                    newLbl));
+            newPnl = new Panel();
+            newPnl.add(newBtn);
+            newPnl.add(newLbl);
+            add(newPnl);
+        }
+    }
+
+    protected class DoneLabel extends Label {
+        public DoneLabel() {
+            super("Done");
+            setVisible(false);
+        }
+    }
+
+    protected class IconTestActionListener implements ActionListener {
+        IconTestFrame f;
+        DoneLabel l;
+
+        public IconTestActionListener(IconTestFrame frame, DoneLabel label) {
+            this.f = frame;
+            this.l = label;
+        }
+
+        public void actionPerformed(ActionEvent e) {
+            f.pack();
+            f.setVisible(true);
+            l.setVisible(true);
+            IconTestButtons.this.pack();
+        }
+    }
+}
+
+class IconTestFrame extends Frame {
+    private String testName;
+    int width, height;
+    Image iconImage;
+    MediaTracker tracker;
+
+    public IconTestFrame(String testName, String iconFileName) {
+        super(testName);
+        this.testName = testName;
+        tracker = new MediaTracker(this);
+
+        //Set icon image
+        URL url = MegaIconTest.class.getResource(iconFileName);
+        Toolkit tk = Toolkit.getDefaultToolkit();
+        if (tk == null) {
+            System.out.println("Toolkit is null!");
+        }
+        if (url == null) {
+            System.out.println("Can't load icon is null!");
+            return;
+        }
+        try {
+            iconImage = tk.createImage((ImageProducer) url.getContent());
+        } catch (java.io.IOException e) {
+            System.out.println("Unable to load icon image from url: " + url);
+        }
+        tracker.addImage(iconImage, 0);
+        try {
+            tracker.waitForAll();
+        } catch (java.lang.InterruptedException e) {
+            System.err.println(e);
+        }
+        width = iconImage.getWidth(this);
+        height = iconImage.getHeight(this);
+        setIconImage(iconImage);
+
+        addWindowListener(new WindowAdapter() {
+            public void windowClosing(WindowEvent e) {
+                setVisible(false);
+            }
+        });
+
+        setLayout(new BorderLayout());
+
+        //Add the icon graphic and instructions to the Frame
```

Comment:
The background transparency of different gifs can be checked by setting the background of the Frame to a different color.  For instance` setBackground(Color.YELLOW)` here and with the original test.

Adding a different background color instead of White makes it easier for the tester to compare the different type of icons.

Reply:
I meant setting it to `setBackground(Color.YELLOW)` instead of default white background at Line #193.

You might have missed this comment as to why I suggested the above - 
Does the test mean "icon with opaque background" by "opaque icon"? If yes, then we should use one .gif with opaque background and another with transparent background. Currently both gif files - dukeWave.gif and fight.gif, have transparent backgrounds.

Reply:
Yeah, had missed this point. Updated now, thanks for pointing it out.

Reply:
This update looks good. But we still need one .gif that has opaque background.

Reply:
Yeah, updated with opaque image, `duke_404.gif`.